src/microbe_stage/editor/CellEditorComponent.GUI.cs

```diff
@@ -90,6 +90,25 @@ private void SetMembraneTooltips(MembraneType referenceMembrane)
             var tooltip = GetSelectionTooltip(membraneType.InternalName, "membraneSelection");
             tooltip?.WriteMembraneModifierList(referenceMembrane, membraneType);
         }
+
+        // Osmoregulation cost is based on the membrane, so update the osmoregulation tooltips
+        // after updating the membrane
+        SetOsmoregulationTooltips();
+    }
+
+    private void SetOsmoregulationTooltips()
+    {
+        var organelles = SimulationParameters.Instance.GetAllOrganelles();
+
+        foreach (var organelle in organelles)
+        {
+            float osmoregulationCost = organelle.HexCount * Membrane.OsmoregulationFactor
+                * Constants.ATP_COST_FOR_OSMOREGULATION
+                * Editor.CurrentGame.GameWorld.WorldSettings.OsmoregulationMultiplier;
```

Comment:
You should extract this osmoregulation multiplier as a variable here as this needs to look up a ton of object properties to follow the chain to the data on each loop iteration.